### PR TITLE
Random Battles winrates: Fine-tune collection

### PR DESCRIPTION
**Pull request by livid-washed**
This fuses together some identical Pokemon formes, and separates formes that are sufficiently different.

Fuses together: Dudunsparce, Maushold, Tatsugiri, Squawkabilly-Blue with Squawkabilly, Squawkabilly-White with Squawkabilly-Yellow.

Separates: Gourgeist, Zacian, Zamazenta, Kyogre/Groudon, megas.

### CODE REVIEW

File: server/chat-plugins/randombattles/winrates.ts [hunk 1]
```diff
@@ -88,10 +101,36 @@ function getSpeciesName(species: string) {
...
+		return "Castform-Snowy";
+	} else if (species === "Cherrim" && moves.includes('Sunny Day')) {
+		return "Cherrim-Sunshine";
+	} else if (species === "Meloetta" && moves.includes('Relic Song')) {
+		return "Meloetta-Pirouette";
+	} else if (species === "Rayquaza" && moves.includes('Dragon Ascent') && megaRayquazaPossible) {
+		return "Rayquaza-Mega";
+	} else if (species === "Greninja" && ability === 'Battle Bond' && ['gen6', 'gen7'].includes(format.mod)) {
+		return "Greninja-Ash";
+	} else if (species === "Aegislash" && moves.includes('Iron Head')) {
```
[the remark this answers is not in the document]
Comment (livid-washed) [reply server/chat-plugins/randombattles/winrates.ts:132]: Regular Aegislash doesn't always roll King's Shield as a move in randbats, so using that to distinguish the sets doesn't always work. Looking at Steel STAB (Flash Cannon for regular / Iron Head for Blade) are more reliable for separating them.

[the remark this answers is not in the document]
Comment (livid-washed) [reply server/chat-plugins/randombattles/winrates.ts:132]: No, but if they end up with drastically different winrates then rands staff could consider reworking/removing one of them

[the remark this answers is not in the document]
Comment (livid-washed) [reply server/chat-plugins/randombattles/winrates.ts:132]: will make it not apply to gen 9 for futureproofing, though

**Comment by MathyFurret** [comment]
Functionally this looks good but was separating all these in-battle formes agreed upon? Unless they're going to be tiered separately it just seems a bit arbitrary to me. I only see a few messages on Discord regarding this

**Comment by ACakeWearingAHat** [comment]
I think it'll be best on further review to revert that bit, even if I approved it initially.

**Comment by livid-washed** [comment]
Alright, let's revert it, no problem

**Review by ACakeWearingAHat — APPROVED** [review]

**Pull request merged**